File: StudyPlans/DataStructure_I/1.TwoSum.py
# Learning enumerate loop
def twoSum(nums):
  for index, value in enumerate(nums):
    print(index, value)    

twoSum([2,7,11,15])

import json

class Solution:
  def twoSum(nums, target):
    hashTable = {}
    for index, value in enumerate(nums):
      hashTable[value] = index
      difference = target - value
      if difference in hashTable:
        return print([hashTable[difference], index])
    
    print(json.dumps(hashTable, indent=4))

  twoSum([2,7,11,15], 9)

File: StudyPlans/DataStructure_I/1.TwoSum.ts
function twoSum(nums: number[], target: number): number[] | void {
  const hashTable: any = {}

  // Equivalent of python enumerate()
  for (const [index, value] of nums.entries()) {
    const difference = target - value
    
    if (difference in hashTable) {
      console.log([hashTable[difference], index])
      return [hashTable[difference], index]
    }
    
    hashTable[value] = index;

    console.log(`${index}: ${value}`);
  }

  return console.error(`Out of reach, invalid target`);
};

twoSum([3, 2, 4], 6)